drivers/sensor/MLX90640.zig

```diff
@@ -0,0 +1,882 @@
+                self.params.kv[i] = @intFromFloat(temp - 0.5);
+            } else {
+                self.params.kv[i] = @intFromFloat(temp + 0.5);
+            }
+        }
+
+        self.params.kvScale = kvScale;
+    }
+
+    fn extractCilc(self: *Self) void {
+        var offsetSP: [2]i16 = undefined;
+        var alphaSP: [2]f32 = undefined;
+        const alphaScale: u8 = @intCast(((self.eeprom[32] & 0xF000) >> 12) + 27);
+
+        offsetSP[0] = @intCast(self.eeprom[58] & 0x03FF);
+        if (offsetSP[0] > 511) {
+            offsetSP[0] = offsetSP[0] - 1024;
+        }
+
+        offsetSP[1] = @intCast((self.eeprom[58] & 0xFC00) >> 10);
+        if (offsetSP[1] > 31) {
+            offsetSP[1] = offsetSP[1] - 64;
+        }
+        offsetSP[1] = offsetSP[1] + offsetSP[0];
+
+        alphaSP[0] = @floatFromInt(self.eeprom[57] & 0x03FF);
+        if (alphaSP[0] > 511) {
+            alphaSP[0] = alphaSP[0] - 1024;
+        }
+        alphaSP[0] = alphaSP[0] / std.math.pow(f32, 2, @floatFromInt(alphaScale));
+
+        alphaSP[1] = @floatFromInt((self.eeprom[57] & 0xFC00) >> 10);
+        if (alphaSP[1] > 31) {
+            alphaSP[1] = alphaSP[1] - 64;
+        }
+        alphaSP[1] = (1 + alphaSP[1] / 128) * alphaSP[0];
+
+        self.params.cpKta = @floatFromInt(self.eeprom[59] & 0x00FF);
+        if (self.params.cpKta > 127) {
+            self.params.cpKta = self.params.cpKta - 256;
+        }
+        const ktaScale1: u8 = @intCast(((self.eeprom[56] & 0x00F0) >> 4) + 8);
+        self.params.cpKta /= std.math.pow(f32, 2, @floatFromInt(ktaScale1));
+
+        var cpKv: f32 = @floatFromInt((self.eeprom[59] & 0xFF00) >> 8);
+        if (cpKv > 127) {
+            cpKv = cpKv - 256;
+        }
+        const kvScale: u8 = @intCast((self.eeprom[56] & 0x0F00) >> 8);
+        self.params.cpKv = cpKv / std.math.pow(f32, 2, @floatFromInt(kvScale));
+
+        self.params.cpAlpha[0] = alphaSP[0];
+        self.params.cpAlpha[1] = alphaSP[1];
+        self.params.cpOffset[0] = offsetSP[0];
+        self.params.cpOffset[1] = offsetSP[1];
+    }
+
+    fn extractDeviatingPixels(self: *Self) i16 {
+        var pixCnt: u32 = 0;
+        for (0..5) |i| {
+            pixCnt = @intCast(i);
+            self.params.brokenPixels[pixCnt] = 0xFFFF;
+            self.params.outlierPixels[pixCnt] = 0xFFFF;
+        }
+
+        var brokenPixCnt: u16 = 0;
+        var outlierPixCnt: u16 = 0;
+        pixCnt = 0;
+        while (pixCnt < 768 and brokenPixCnt < 5 and outlierPixCnt < 5) {
+            if (self.eeprom[pixCnt + 64] == 0) {
+                self.params.brokenPixels[brokenPixCnt] = @intCast(pixCnt);
+                brokenPixCnt = brokenPixCnt + 1;
+            } else if ((self.eeprom[pixCnt + 64] & 0x0001) != 0) {
+                self.params.outlierPixels[outlierPixCnt] = @intCast(pixCnt);
+                outlierPixCnt = outlierPixCnt + 1;
+            }
+
+            pixCnt = pixCnt + 1;
+        }
+
+        var warn: i16 = 0;
```

Comment:
We don't need to use numbers here... this isn't C :)

Reply:
Oh yeah, I was copying so much c++ this was bound to happen.